flow/directives.py

```diff
@@ -255,32 +255,36 @@ def _evaluate(value, jobs):
 
 
 class _OnlyType:
-    def __init__(self, type_, preprocess=None, postprocess=None):
+    def __init__(self, *types, preprocess=None, postprocess=None):
         def identity(value):
             return value
 
-        self.type = type_
+        self.types = types
         self.preprocess = identity if preprocess is None else preprocess
         self.postprocess = identity if postprocess is None else postprocess
 
     def __call__(self, value):
         return self.postprocess(self._validate(self.preprocess(value)))
 
     def _validate(self, value):
-        if isinstance(value, self.type):
+        if isinstance(value, self.types):
             return value
-        try:
-            return self.type(value)
-        except Exception:
-            raise TypeError(
-                f"Expected an object of type {self.type}. "
-                f"Received {value} of type {type(value)}."
-            )
-
-
-def _raise_below(threshold):
+        for type_ in self.types:
```

Comment:
I am not really satisfied with the logic I used here. But since it's internal and the variables which use this are internal as well, I am not really sure if we should be too worried about this.

Reply:
I think this code is fine. It looks identical to the place in hoomd-blue where I assume @b-butler [originally advocated copying it from](https://github.com/glotzerlab/hoomd-blue/blob/master/hoomd/data/typeconverter.py#L153).